app/views/courses/_block.json.jbuilder

```diff
@@ -19,7 +19,11 @@ if block.training_modules.any?
     json.due_date due_date_manager.computed_due_date.strftime('%Y/%m/%d')
     json.overdue due_date_manager.overdue?
     json.deadline_status due_date_manager.deadline_status
-    json.flags due_date_manager.flags
+    if tm.exercise? && !due_date_manager.flags.nil?
```

Comment:
I don't like having this logic in the view. I think it would be better to change the due date manager `flags` method so that it takes an option course_id param.

Reply:
Okay! I will move the logic inside due date manager flags method.